apps/projects/app/components/Panel/NewProject/RepoSelector.js

```diff
@@ -85,50 +91,54 @@ class Repo extends React.Component {
 
   unselect = () => this.setState({})
 
+  repoList = (visibleRepos, repoArray) => visibleRepos.length > 0 ? (
+    <RadioList
+      items={repoArray}
+      selected={this.state.repoSelected}
+      onChange={this.onRepoSelected}
+    />
+  ) : (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        marginTop: '19px',
+        flexDirection: 'column',
+      }}
+    >
+      <Text.Block>
+        There are no repositories matching{' '}
+        <Text weight="bold">{this.state.filter}</Text>
+      </Text.Block>
+      <ClearSearch onClick={this.handleClearSearch}>
+        Clear Search
+      </ClearSearch>
+    </div>
+  )
+
+  stillLoading = () => !(this.props.data && this.props.data.viewer && this.props.data.viewer.repositories.edges.length > 0)
```

Comment:
Can you try and reduce the line length here? This is still hard to read...

Reply:
Improved, although it still looks not very smart. There must be a better way of chaining conditions like that: if X and if X.Y and if X.Y.Z...